src/main/scala/com/lightbend/lagom/core/persistence/effects/core/persistence/AccountEntity.scala

```diff
@@ -28,6 +28,19 @@ class AccountEntity extends PersistentEntity[AccountCommand[_], AccountEvent, Ac
       .andThen { // Some[State] => Actions == State => Actions
         // here come the actions that are valid when Account
         // is already created
+        //
+        // IMS: I think it is unnecessary to make the argument of `andThen` a PartialFunction. If this branch
+        // is equivalent to `State => Actions` then we could require a `Function[State, Actions]`, right?
+        //
+        //
+        //     Behavior
+        //        .first { depositCommandActions orElse atCreationEvents}
+        //        .andThen { account =>
+        //              readOnlyCommands orElse
+        //              moreCommands orElse
+        //                  ...
+        //
+        //
```

Comment:
That's true, but ...

A `PartialFunction` is legal where `Function` is required. 

If the user call it using `PartialFunction` syntax:
```
andThen {
 case acc: OpenAccount => ...
 case acc: ClosedAccount => ...
```

That will also work and if there is a State that was not matched we will get a match error. 

I didn't code that yet, but the goal is to take the `PF` passed by the user and compose with a catch all.
```
userPf
 .orElse { 
  case state => throw new NoHandlersFoundException("No handlers found for current state, check the completeness of your behavior definition" )
}
```

This is only to avoid the `MatchError`. 

Alternatively we can wrap on a `Try` and recover with an exception that gives more context.